### PR TITLE
Show linked documents

### PR DESCRIPTION
**Pull request by pezholio**
This builds on https://github.com/alphagov/publishing-api/pull/2823 to allow us to fetch linked documents for a given content block. This looks for any documents linked to the Content Block with the link type `embed` and lists the documents, together with their type and organisation (if we can find it)

## Screenshot

image

### CODE REVIEW

File: lib/engines/content_object_store/test/unit/app/services/get_linked_content_items_test.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      result = ContentObjectStore::GetLinkedContentItems.new(content_block_document:, page: 1).items
+
+      assert_equal result[0].title, response["results"][0]["title"]
+      assert_equal result[0].base_path, response["results"][0]["base_path"]
+      assert_equal result[0].document_type, response["results"][0]["document_type"]
+      assert_equal result[0].organisation, organisations.first
+    end
+
+    it "calls the client with the expected arguments" do
+      ContentObjectStore::GetLinkedContentItems.new(content_block_document:, page: 1).items
```
Comment (Harriethw) [inline lib/engines/content_object_store/test/unit/app/services/get_linked_content_items_test.rb:70]: is this missing an assert on the publishing api call?

File: lib/engines/content_object_store/app/views/content_object_store/content_block/documents/show.html.erb [hunk 2]
```diff
@@ -12,6 +12,30 @@
...
+    before the table, the existing component doesn't support that so I've left
+    it out for now-->
+    <%= render(
+      ContentObjectStore::ContentBlock::Document::Show::LinkedEditionsTableComponent.new(
+        caption: "Content appears in",
+        linked_content_items: @linked_content_items,
+      ),
+    ) %>
+
+    <% if @page_data.total_pages > 1 %>
```
Comment (Harriethw) [inline lib/engines/content_object_store/app/views/content_object_store/content_block/documents/show.html.erb:27]: should this be part of the LinkedEditionsTable? 
Should we have some testing around pagination as well? It's a bit hard to manually test without creating a bunch of links... (perhaps seeded test data might be helpful at this point too?)

**Review by Harriethw — APPROVED** [review]
Looks great! 
Just a Q around the pagination, but I guess as we are still in prototyping mode it may not be necessary.

I was able to get some Links working locally after a bit of trial and error with the following steps:
1. Add `content_block_email_address` to the `SUPPORTED_DOCUMENT_TYPES` in Publishing API's `EmbeddedContentFinderService`.
2. Create a published Content Block in Whitehall. Grab it's content_id from a console. `content_block_id = ContentObjectStore::ContentBlock::Document.find({document_id}).content_id` 
3. Create a new Document in Whitehall - I used Publication. and put in the body the embed with your block content_id `{{embed:content_block_email_address:400420d3-054b-4542-9d6e-f37bef97800b}}`. Save the Document.
4. find that Document's content_id `document_id = Document.find(Edition.last.document_id).content_id`
4. Publish that document in the Whitehall console, using the doucment id `Services.publishing_api.publish("4083a475-7161-465c-871d-9e77b91f2694", "major")` 
5. check in Publishing API that it has the `embed` link http://publishing-api.dev.gov.uk/v2/content/{{document_id}}
6. view Links on the Content Block

File: lib/engines/content_object_store/app/views/content_object_store/content_block/documents/show.html.erb (hunk 2 above)
Comment (pezholio) [reply lib/engines/content_object_store/app/views/content_object_store/content_block/documents/show.html.erb:27]: Good idea! The pagination is tested in the Cucumber tests, but I think it's a good idea to roll the pagination into the component

Comment (pezholio) [reply lib/engines/content_object_store/app/views/content_object_store/content_block/documents/show.html.erb:27]: I'll merge now, but we can maybe look at improving this as part of https://trello.com/c/2sFJMLTO/364-show-what-pages-will-be-affected-by-a-content-change-before-confirming-the-change

**Pull request merged**